Formula/go.rb

```diff
@@ -24,20 +24,20 @@ class Go < Formula
   resource "gobootstrap" do
     on_macos do
       if Hardware::CPU.arm?
-        url "https://storage.googleapis.com/golang/go1.16.darwin-arm64.tar.gz"
-        version "1.16"
-        sha256 "4dac57c00168d30bbd02d95131d5de9ca88e04f2c5a29a404576f30ae9b54810"
+        url "https://golang.org/dl/go1.17.darwin-arm64.tar.gz"
```

Comment:
I pulled these new URLs from https://golang.org/dl/. It looks like the same URL exists on storage.googleapis.com if we wanted to keep it the same, but I figured I'd update it to the link in the official docs instead.